burnman/layer.py

```diff
@@ -0,0 +1,797 @@
+
+        self.name = name
+        self.min_depth = min_depth
+        self.max_depth = max_depth
+        self.thickness = self.max_depth - self.min_depth
+        self.n_slices = n_slices
+        self.depths = np.linspace(min_depth, max_depth, n_slices)
+        self.radius_planet = radius_planet
+        self.radii = self.radius_planet - self.depths
+        self.outer_radius = max(self.radii)
+        self.inner_radius = min(self.radii)
+        self.verbose = verbose
+        self._cached = {}
+        self._pressures = None
+        self._temperatures = None
+        self.sublayers = None
+    
+    def reset(self):
+        """
+        Resets all cached material properties.
+
+        It is typically not required for the user to call this function.
+        """
+        self._cached = {}
+        self._pressures = None
+        self._temperatures = None
+        self.sublayers = None
+
+    def set_material(self, material):
+        """
+        Set the material of a Layer with a Material
+        """
+        assert(isinstance(material, Material))
+        self.material = material
+        self.reset()
+
+    def set_temperature_mode(self, temperature_mode='adiabat',
+                             temperatures=None, temperature_top=None):
+        """
+        Sets temperature of the layer by user-defined values or as an (modified) adiabat.
+        temperature_mode is 'user_defined','adiabatic', or 'modified_adiabat'
+
+        Parameters
+        ----------
+        temperature_mode : string
+        This can be set to 'user_defined','adiabatic', or 'modified_adiabat'
+        temperatures : array of float
+        The desired fixed temperatures in [K]. Should have same length as defined depths in layer.
+        temperature_top : float
+        Temperature at the top for an adiabat
+        
+        Note
+        ---------
+        'user_defined' = fixes the temperature with the profile input by the users
+        'adiabat' = self-consistently computes the adiabat when setting the state of the layer
+        'modified_adiabat' = adds the user input array to the adiabat, 
+            e.g. allows to implement boundary layers
+        """
+        self.reset()
+        assert(temperature_mode == 'user_defined' or temperature_mode ==
+               'adiabat' or temperature_mode == 'modified_adiabat')
+
+        self.temperature_mode = temperature_mode
+
+        if temperature_mode == 'user_defined' or temperature_mode == 'modified_adiabat':
+            assert(len(temperatures) == len(self.depths))
+            self.usertemperatures = temperatures
+        else:
+            self.usertemperatures = np.zeros_like(self.depths)
+
+        if temperature_mode == 'adiabat' or temperature_mode == 'modified_adiabat':
+            self.temperature_top = temperature_top
+        else:
+            self.temperature_top = None
+
+    def set_state( self, pressure_mode='selfconsistent', pressures=None, pressure_top=None,
```

Comment:
I would prefer 'self-consistent'. Similarly, I think I prefer 'user-defined' for the temperature_mode and pressure_mode names.